src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -679,11 +679,18 @@ public static void Beep()
         {
             if (!Console.IsOutputRedirected)
             {
-                ReadOnlySpan<byte> bell = "\u0007"u8; // Windows doesn't use terminfo, so the codepoint is hardcoded.
-                int errorCode = WindowsConsoleStream.WriteFileNative(OutputHandle, bell, useFileAPIs: Console.OutputEncoding.CodePage != UnicodeCodePage);
-                if (errorCode == Interop.Errors.ERROR_SUCCESS)
+                Span<byte> bell = stackalloc byte[10];
+                if (Console.OutputEncoding.TryGetBytes("\u0007", bell, out int bytesWritten))
```

Comment:
`OutputEncoding` can be a custom encoding provided by the user. Nothing guarantees that it is going to encode the bell character into less than 10 bytes. For example, the encoding can be used as a hook to introduce some ridiculous escape sequences.

I would keep this simple a just let the encoding to allocate the array.